src/tar-stream-to-objects.js

```diff
@@ -5,28 +5,36 @@ const Readable = require('readable-stream')
 
 // transform tar stream into readable stream of
 // { path: 'string', content: Readable }
-module.exports = function (err, res, send, done) {
+module.exports = (err, res, send, done) => {
   if (err) {
     return done(err)
   }
 
-  var ex = tar.extract()
+  const ex = tar.extract()
   res.pipe(ex)
 
-  var objStream = new Readable({ objectMode: true })
+  const objStream = new Readable({ objectMode: true })
   objStream._read = function noop () {}
 
-  ex.on('entry', function (header, stream, next) {
-    objStream.push({
-      path: header.name,
-      content: header.type !== 'directory' ? stream : null
-    })
-    next()
+  ex.on('entry', (header, stream, next) => {
+    stream.on('end', next)
```

Comment:
This has to happen already if you read the documentation of tar-stream. This only makes sure that we actually process things correctly. The current implementation throws up and dies on larger sets of files.